### PR TITLE
Nonamr merge

### PR DESCRIPTION
**Pull request by tkoskela**
Merged AMR and non-AMR solver branches. AMR solver gets called only when Parameters::amrMaxSpatialRefLevel > 0.

### CODE REVIEW

File: spatial_cell.cpp [hunk 1]
```diff
@@ -654,7 +654,7 @@ namespace spatial_cell {
             // Send this data only to ranks that contain face neighbors
             // this->neighbor_number_of_blocks has been initialized to 0, on other ranks it can stay that way.
             const set<int>& ranks = this->face_neighbor_ranks[neighborhood];
-            if ( receiving || ranks.find(receiver_rank) != ranks.end()) {
+            if ( P::amrMaxSpatialRefLevel == 0 || receiving || ranks.find(receiver_rank) != ranks.end()) {
```
Comment (ykempf) [inline spatial_cell.cpp:657]: So this has no alternative path for >0?

File: vlasovsolver/cpu_trans_map.cpp [hunk 2]
```diff
@@ -654,66 +653,55 @@ void update_remote_mapping_contribution(
    int myRank;
    
    MPI_Comm_rank(MPI_COMM_WORLD,&myRank);
+
```
Comment (ykempf) [inline vlasovsolver/cpu_trans_map.cpp:655]: This could be in a timed barrier?

File: vlasovsolver/cpu_trans_map.cpp [hunk 3]
```diff
@@ -797,7 +785,10 @@ void update_remote_mapping_contribution(
    for (size_t c=0; c < receiveBuffers.size(); ++c) {
       aligned_free(receiveBuffers[c]);
    }
-}
 
-*/
+   // MPI_Barrier(MPI_COMM_WORLD);
```
Comment (ykempf) [inline vlasovsolver/cpu_trans_map.cpp:788]: Same here, could be put in a timed barrier.

File: spatial_cell.cpp (hunk 1 above)
Comment (tkoskela) [reply spatial_cell.cpp:657]: It does, the alternative path is everything to the right of the first ||. I had to add the == 0 case to make it work when there is no AMR.

Comment (ykempf) [reply spatial_cell.cpp:657]: O'rite.

File: vlasovsolver/cpu_trans_map.cpp (hunk 2 above)
Comment (tkoskela) [reply vlasovsolver/cpu_trans_map.cpp:655]: That MPI_Comm_rank call is probably a debugging remnant, it's not used anywhere. Why do we want a barrier?

Comment (ykempf) [reply vlasovsolver/cpu_trans_map.cpp:655]: No I mean the commented block after that.

Comment (tkoskela) [reply vlasovsolver/cpu_trans_map.cpp:655]: Well, that is also left overs for my debugging. Basically I was printing out stuff from get_mpi_datatype() in spatialCell and I had to mark where the output from update_remote_mapping_contribution() starts and ends in order to parse the data with a script. I can do a round of cleanup of commented code if you want.

Comment (ykempf) [reply vlasovsolver/cpu_trans_map.cpp:655]: As you like, I just looked at the diffs and commented on what made sense (or didn't quite) to me.

**Comment by ykempf** [comment]
Overall I'm fine with this now, I didn't see anything eyebrow-raising.

**Pull request merged**